apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainProcessorImpl.java

```diff
@@ -1069,10 +1086,9 @@ private void accept(Block block, List<Transaction> validPhasedTransactions, List
             }
             log.debug("Total rollback time: {} ms", System.currentTimeMillis() - rollbackStartTime);
             dataSource.clearCache();
-            dataSource.commit(false); // should happen definately, otherwise
-
-        }
-        catch (RuntimeException e) {
+            invalidateAllPublicKeyCache();
```

Comment:
Such invalidation makes publicKeyCache useless.

Reply:
That was en experiment to check suspicion which was confirmed. Now I commented that out because we need more so sophisticated solution.